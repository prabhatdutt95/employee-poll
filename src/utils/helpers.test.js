import { formatDate } from "./helpers"; // adjust the import path as necessary

describe("formatDate", () => {
  it("returns the time in the correct format", () => {
    // This is a fixed timestamp for testing (January 1, 1970, 00:00:00 UTC)
    const timestamp = 0;
    const result = formatDate(timestamp);
    // Expected format "HH:MM AM/PM | MM/DD/YYYY"
    expect(result).toMatch(/^\d{1,2}:\d{2}:AM \| \d{1,2}\/\d{1,2}\/\d{4}$/);
  });

  it("handles PM times correctly", () => {
    // Timestamp for January 1, 1970, 13:00:00 UTC (1:00 PM UTC)
    const timestamp = 13 * 60 * 60 * 1000;
    const result = formatDate(timestamp);
    expect(result).toContain("PM"); // Check if 'PM' is included for times after noon
    expect(result).toMatch(/^\d{1,2}:\d{2}:PM \| \d{1,2}\/\d{1,2}\/\d{4}$/); // Complete format check
  });

  it("handles AM times correctly", () => {
    // Timestamp for January 1, 1970, 01:00:00 UTC (1:00 AM UTC)
    const timestamp = 1 * 60 * 60 * 1000;
    const result = formatDate(timestamp);
    expect(result).toContain("AM"); // Check if 'AM' is included for times before noon
    expect(result).toMatch(/^\d{1,2}:\d{2}:AM \| \d{1,2}\/\d{1,2}\/\d{4}$/); // Complete format check
  });

  it("works correctly with different dates", () => {
    // Timestamp for a known date (e.g., July 20, 1969, 20:18 UTC)
    const timestamp = new Date("July 20, 1969, 13:18:00 UTC").getTime();
    const result = formatDate(timestamp);
    // Ensure the date part of the string matches the known date
    expect(result).toContain("1969");
    expect(result).toMatch(/^\d{1,2}:\d{2}:PM \| \d{1,2}\/\d{1,2}\/\d{4}$/); // Complete format check
  });

  it("creates a consistent date snapshot", () => {
    const timestamp = 0; // January 1, 1970, 00:00:00 UTC
    const formattedDate = formatDate(timestamp);
    expect(formattedDate).toMatchSnapshot();
  });
});
