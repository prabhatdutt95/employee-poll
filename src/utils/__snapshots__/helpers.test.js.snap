// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatDate creates a consistent date snapshot 1`] = `"5:30:AM | 1/1/1970"`;
